scripts/stablex/setup_thegraph_data.js

```diff
@@ -0,0 +1,233 @@
+const { solutionSubmissionParams, generateTestCase } = require("../../test/resources/examples")
+
+const { makeDeposits, placeOrders } = require("../../test/stablex/stablex_utils")
+const { closeAuction } = require("../../scripts/stablex/utilities.js")
+const { decodeOrdersBN } = require("../../src/encoding")
+
+const { toETH, feeAdded, feeSubtracted, ERROR_EPSILON } = require("../../test/resources/math")
+const { waitForNSeconds } = require("../../test/utilities")
+
+const BatchExchange = artifacts.require("BatchExchange")
+const ERC20Mintable = artifacts.require("ERC20Mintable.sol")
+
+async function _getOwl() {
+  const TokenOWL = artifacts.require("TokenOWL")
+  const TokenOWLProxy = artifacts.require("TokenOWLProxy")
+  const owlProxyContract = await TokenOWLProxy.deployed()
+  return TokenOWL.at(owlProxyContract.address)
+}
+
+async function _mintOWL({ owl, user, amount, batchExchange }) {
+  console.log("Mint %d of OWL for user %s (%d)", amount, user.address, user.id)
+  await owl.mintOWL(user.address, amount)
+  await owl.approve(batchExchange.address, amount)
+}
+
+async function _createAndAddToken(batchExchange) {
+  const token = await ERC20Mintable.new()
+  await batchExchange.addToken(token.address)
+  const tokenId = await batchExchange.tokenAddressToIdMap(token.address)
+  const id = tokenId.toNumber()
+
+  console.log("Added token %s with id %d", token.address, id)
+  return {
+    id,
+    address: token.address,
+    instance: token,
+  }
+}
+
+async function _mintToken({ token, user, amount, batchExchange }) {
+  console.log("Mint token %d of token %s for user %s (%d)", amount, token.address, user.address, user.id)
+  await token.instance.mint(user.address, amount)
+  await token.instance.approve(batchExchange.address, amount, { from: user.address })
+}
+
+async function _getBatchId(batchExchange) {
+  const batchId = await batchExchange.getCurrentBatchId.call()
+  return batchId.toNumber()
+}
+
+function _getTestCase({ user1, user2, token1 }) {
```

Comment:
Do you want to use this function in the future in different ways? If not, if it is really only used once, you could just get the example directly from here
 https://github.com/gnosis/dex-contracts/blob/master/test/resources/examples/index.js#L7
and reduce code duplication.

Reply:
I didn't do it cause that code was relying on a clean newly deployed contract (no previous tokens had been listed, so we use token 0 and 1)

So in my approach I was planning to rely only on my data. Anyways, u are right. I'll try to reuse this cases.